### PR TITLE
chore(settings): prepare prod and stage configs for deployment

### LINKED ISSUE #24: Set rocket config for stage and prod
There is a config file for Rocket in the root directory, `Rocket.toml`. Most of the stuff in there is pretty standard, address and port to listen on, etc. You can read more about it at:

https://api.rocket.rs/rocket/config/

One that stands out as interesting for prod is `workers` / `ROCKET_WORKERS`, which defines how many worker threads Rocket should create for handling requests. The default value is CPU count * 2. We'll probably need to do some testing to zero in on the optimal value for that, especially given that it will be sharing a box with auth server and auth db processes.

### LINKED ISSUE #100: Do not run in production without a changed message_id_hmac_key
We should panic or something if we run the service in production and it does not the `message_id_hmac_key` value changed

### PR DESCRIPTION
**Pull request by brizental**
Fixes #100 
Fixes #24 

As mentioned in #24 we still need to see about `ROCKET_WORKERS`.

Also, in production, rocket sends a warning about not having set a secret_key, but I decided to leave it without anything for now. Anyways that would be something that probably we wouldn't "hard code" here.

r? @vladikoff @philbooth

### CODE REVIEW

File: src/settings/mod.rs [hunk 1]
```diff
@@ -130,11 +130,18 @@ impl Settings {
...
 
         config.merge(File::with_name("config/local").required(false))?;
-
         config.merge(Environment::with_prefix("fxa_email"))?;
 
         match config.try_into::<Settings>() {
             Ok(settings) => {
+                if let Ok(rocket_env) = env::var("ROCKET_ENV") {
+                    if rocket_env == "production"
+                        && &settings.message_id_hmac_key == "YOU MUST CHANGE ME"
```
Comment (brizental) [inline src/settings/mod.rs]: I'm thinking here it would be best if we actually checked that the HMAC was valid, instead of just checking for `YOU MUST CHANGE ME`. But I remember us talking in a meeting about doing it this way, so let me know what you guys think :)

Comment (vladikoff) [reply src/settings/mod.rs]: > 'm thinking here it would be best if we actually checked that the HMAC was valid

Sure let's do that if possible, and we also need a sample one for `dev.json` right?

Comment (brizental) [reply src/settings/mod.rs]: When something is not set in `dev.json` it defaults to `default.json` so in there we would just get `YOU MUST CHANGE ME`. Do you think we should have something else for `dev.json`?

Comment (vladikoff) [reply src/settings/mod.rs]: from mtg: So these can be any string so let's keep it this way
